### PR TITLE
ci.yml: refactor workflow

### PR DESCRIPTION
**Pull request by drakulavich**
* Use setup actions for python and node
* Cache dependencies
* Remove redundant step 'Setup Python for PDF generation'

- [ ] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [ ] The page(s) have at most 8 examples.
- [ ] The page description(s) have links to documentation or a homepage.
- [ ] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [ ] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- **Version of the command being documented (if known):**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -29,27 +36,22 @@ jobs:
     - name: Install npm dependencies
       run: npm ci
 
-    - name: Install python dependencies
-      run: python3 -m pip install -r requirements.txt
+    - name: Install pip dependencies
+      run: |
+        pip install --upgrade pip
```
Comment (MasterOdin) [inline .github/workflows/ci.yml:41]: This currently doesn't do anything, and I don't think we'll need it into the future.

**Review by MasterOdin — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -29,27 +36,22 @@ jobs:
...
-      with:
-        python-version: '3.x'
-
     - name: Build PDF
       if: github.repository == 'tldr-pages/tldr' && github.ref == 'refs/heads/main'
       run: |
         cd scripts/pdf/
-        pip3 install -r requirements.txt
-        python3 render.py ../../pages -c solarized-light
+        python render.py ../../pages -c solarized-light
```
[the remark this answers is not in the document]
Comment (drakulavich) [reply .github/workflows/ci.yml]: Nice one! Thank you 👍

File: .github/workflows/ci.yml (hunk 1 above)
Comment (drakulavich) [reply .github/workflows/ci.yml]: Ok, I removed `pip` upgrade.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -38,7 +33,6 @@ jobs:
 
     - name: Install pip dependencies
       run: |
-        pip install --upgrade pip
         pip install -r requirements.txt -r scripts/pdf/requirements.txt
```
Comment (MasterOdin) [inline .github/workflows/ci.yml:36]:
```suggestion
      run: pip install -r requirements.txt -r scripts/pdf/requirements.txt
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -49,8 +43,8 @@ jobs:
 
     - name: Build PDF
       if: github.repository == 'tldr-pages/tldr' && github.ref == 'refs/heads/main'
+      working-directory: ./scripts/pdf
       run: |
-        cd scripts/pdf/
         python render.py ../../pages -c solarized-light
```
Comment (MasterOdin) [inline .github/workflows/ci.yml:48]:
```suggestion
      run: python render.py ../../pages -c solarized-light
```

**Review by MasterOdin — CHANGES_REQUESTED** [review]
Can get rid of the multiline syntax if we only have one line under it.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (drakulavich) [reply .github/workflows/ci.yml]: Sure, updated

**Review by MasterOdin — APPROVED** [review]

**Comment by drakulavich** [comment]
@navarroaxel please, have a look. I've updated the PR according to the suggestions

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -12,15 +12,17 @@ jobs:
...
 
     steps:
-    - name: Setup Python
-      uses: actions/setup-python@v4
+    - uses: actions/checkout@v3
+
+    - uses: actions/setup-python@v4
       with:
-        python-version: 3.8
+        python-version: '3.10'
```
Comment (navarroaxel) [inline .github/workflows/ci.yml:19]:
```suggestion
        python-version: 3.10
```

**Review by navarroaxel — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -12,15 +12,17 @@ jobs:
...
 
     steps:
-    - name: Setup Python
-      uses: actions/setup-python@v4
+    - uses: actions/checkout@v3
+
+    - uses: actions/setup-python@v4
       with:
-        python-version: 3.8
+        python-version: 3.10
```
Comment (navarroaxel) [inline .github/workflows/ci.yml:19]: Now I understand the quotes
```suggestion
        python-version: '3.10'
```

**Comment by drakulavich** [comment]
Thank you for review and merge 👍

**Pull request merged**